addons/mail/static/src/new/messaging.js

```diff
@@ -276,6 +289,14 @@ export class Messaging {
                         this.createMessage(body, message, thread);
                     }
                     break;
+                case "mail.record/insert":
+                    {
+                        const partnersInPayload = notif.payload.Partner;
+                        partnersInPayload.forEach(({ id, im_status }) => {
+                            this.partners[id].im_status = im_status;
+                        });
```

Comment:
Prefer `for () {}` syntax rather than `.forEach()`, for improved readability

Reply:
It is not itterable and `for of` doesn't work on it and `for in` is kind of bulky. Do you want me to change it anyway?

Reply:
From what I see, this is an array for data related to `im_status`.

It's just that the notification name is quite generic, so it can be an object of an array.
It just so happens that fetched `im_status` are returned as an array for `Partner`.
For other notifications `mail.record/insert`, `notif.payload.Partner` can be `undefined`, or even an object.

So you should make sure that `notif.payload.Partner` exists, and if so you must handle case of a single object or array of objects.

Also if the payload has no `im_status`, you should not update `im_status`. You should also make sure of that in this LOC.